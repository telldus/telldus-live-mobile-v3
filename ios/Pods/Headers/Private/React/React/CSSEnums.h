../../../../../../node_modules/react-native/React/CSSLayout/CSSEnums.h