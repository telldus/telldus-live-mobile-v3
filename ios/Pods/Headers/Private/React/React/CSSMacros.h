../../../../../../node_modules/react-native/React/CSSLayout/CSSMacros.h