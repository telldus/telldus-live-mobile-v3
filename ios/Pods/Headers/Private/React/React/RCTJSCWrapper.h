../../../../../../node_modules/react-native/React/Executors/RCTJSCWrapper.h