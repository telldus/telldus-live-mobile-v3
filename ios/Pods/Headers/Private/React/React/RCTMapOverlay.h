../../../../../../node_modules/react-native/React/Views/RCTMapOverlay.h