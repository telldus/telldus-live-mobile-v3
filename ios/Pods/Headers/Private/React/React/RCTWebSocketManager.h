../../../../../../node_modules/react-native/Libraries/WebSocket/RCTWebSocketManager.h