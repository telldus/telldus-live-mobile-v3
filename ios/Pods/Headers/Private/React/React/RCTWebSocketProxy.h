../../../../../../node_modules/react-native/React/Base/RCTWebSocketProxy.h