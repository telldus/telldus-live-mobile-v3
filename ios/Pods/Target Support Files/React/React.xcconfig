CLANG_CXX_LANGUAGE_STANDARD = c++14
CONFIGURATION_BUILD_DIR = $PODS_CONFIGURATION_BUILD_DIR/React
GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1
HEADER_SEARCH_PATHS = "${PODS_ROOT}/Headers/Private" "${PODS_ROOT}/Headers/Private/React" "${PODS_ROOT}/Headers/Public" "${PODS_ROOT}/Headers/Public/React"
OTHER_LDFLAGS = -l"stdc++" -framework "JavaScriptCore"
PODS_BUILD_DIR = $BUILD_DIR
PODS_CONFIGURATION_BUILD_DIR = $PODS_BUILD_DIR/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
PODS_ROOT = ${SRCROOT}
PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.${PRODUCT_NAME:rfc1034identifier}
SKIP_INSTALL = YES
