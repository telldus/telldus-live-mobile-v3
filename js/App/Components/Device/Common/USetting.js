/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

// @flow

'use strict';

import React from 'react';
import { useSelector } from 'react-redux';

import {
	View,
	Text,
	CheckBoxIconText,
} from '../../../../BaseComponents';

import Theme from '../../../Theme';

const USetting = (props: Object): Object => {
	const {
		isChecked,
		onToggleCheckBox,
		intl,
		option,
	} = props;

	const { layout } = useSelector((state: Object): Object => state.app);

	const {
		radioButtonLabelStyle,
		optionWrapStyle,
		optionButtonCover,
		uCheckBoxIconStyle,
		brandPrimary,
	} = getStyles(layout);


	return (
		<View style={optionButtonCover}>
			<CheckBoxIconText
				isChecked={isChecked}
				iconStyle={{
					...optionWrapStyle,
					...uCheckBoxIconStyle,
					backgroundColor: isChecked ? brandPrimary : '#fff',
					color: isChecked ? '#fff' : 'transparent',
				}}
				textStyle={radioButtonLabelStyle}
				onToggleCheckBox={onToggleCheckBox}
				intl={intl}/>
			<Text style={radioButtonLabelStyle}>
				{option}
			</Text>
		</View>
	);

};

const getStyles = (appLayout: Object): Object => {
	const { height, width } = appLayout;
	const isPortrait = height > width;
	const deviceWidth = isPortrait ? width : height;

	const {
		paddingFactor,
		rowTextColor,
		brandPrimary,
		fontSizeFactorTen,
	} = Theme.Core;

	const padding = deviceWidth * paddingFactor;

	const fontSizeText = deviceWidth * fontSizeFactorTen;

	const outerPadding = padding * 2;

	const checkBoxSpace = padding;

	const totalPaddingU = checkBoxSpace * 11;
	const borderWidthU = 1;
	const size = Math.floor((deviceWidth - (outerPadding + totalPaddingU)) / 10);

	return {
		checkBoxSpace,
		brandPrimary,
		optionButtonCover: {
			flexDirection: 'column',
			alignItems: 'center',
			justifyContent: 'center',
		},
		uCheckBoxIconStyle: {
			fontSize: size * 0.8,
			height: size,
			width: size,
			borderColor: brandPrimary,
			borderWidth: borderWidthU,
		},
		radioButtonLabelStyle: {
			fontSize: fontSizeText,
			color: rowTextColor,
			marginTop: padding,
		},
		rButtonStyle: {
		},
		optionWrapStyle: {
			marginTop: padding,
		},
	};
};

export default React.memo<Object>(USetting);

