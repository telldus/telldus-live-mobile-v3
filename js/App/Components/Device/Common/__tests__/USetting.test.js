/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope this it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */


import React from 'react';
import {
	Dimensions,
} from 'react-native';
import renderer from 'react-test-renderer';
import * as redux from 'react-redux';

import USetting from '../USetting';

let {height, width} = Dimensions.get('window');

const spy = jest.spyOn(redux, 'useSelector');
spy.mockReturnValue(
	{
		layout: {
			height,
			width,
		},
	}
);

const intl = {
	formatMessage: () => '',
};

jest.useFakeTimers();

it('renders USetting when checked using Snapshots', () => {
	expect(renderer.create(
		<USetting
			isChecked
			intl={intl}/>
	)).toMatchSnapshot();
});

it('renders USetting when unchecked using Snapshots', () => {
	expect(renderer.create(
		<USetting
			intl={intl}/>
	)).toMatchSnapshot();
});
