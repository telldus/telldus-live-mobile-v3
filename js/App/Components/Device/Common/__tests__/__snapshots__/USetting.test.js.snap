// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders USetting when checked using Snapshots 1`] = `
<View
  colorScheme="light"
  colors={
    Object {
      "ShadeOne": "#eeeeee",
      "ShadeThree": "#8e8e93",
      "ShadeTwo": "#b5b5b5",
      "activeTintOne": "#e26901",
      "background": "#fff",
      "border": "#C9C9CE",
      "card": "#fff",
      "inActiveTintOne": "#e26901",
      "primary": "#1b365d",
      "screenBackground": "#eeeeee",
      "text": "#A59F9A",
      "textFive": "#555555",
      "textFour": "#4C4C4C",
      "textThree": "#000",
      "textTwo": "#777777",
    }
  }
  dark={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityLabel=" , , "
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 2,
        "borderTopLeftRadius": 2,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "overflow": "hidden",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      colorScheme="light"
      colors={
        Object {
          "ShadeOne": "#eeeeee",
          "ShadeThree": "#8e8e93",
          "ShadeTwo": "#b5b5b5",
          "activeTintOne": "#e26901",
          "background": "#fff",
          "border": "#C9C9CE",
          "card": "#fff",
          "inActiveTintOne": "#e26901",
          "primary": "#1b365d",
          "screenBackground": "#eeeeee",
          "text": "#A59F9A",
          "textFive": "#555555",
          "textFour": "#4C4C4C",
          "textThree": "#000",
          "textTwo": "#777777",
        }
      }
      dark={false}
      importantForAccessibility="yes"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#999",
            "fontSize": 30,
          },
          Object {
            "backgroundColor": "#1b365d",
            "borderColor": "#1b365d",
            "borderRadius": 2,
            "borderWidth": 1,
            "color": "#fff",
            "fontSize": 37.6,
            "height": 47,
            "marginTop": 20.83275,
            "overflow": "hidden",
            "padding": 1.4000000000000001,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 47,
          },
          Object {
            "fontFamily": "telldusicons",
          },
        ]
      }
    >
      checkmark
    </Text>
  </View>
  <Text
    allowFontScaling={false}
    colorScheme="light"
    colors={
      Object {
        "ShadeOne": "#eeeeee",
        "ShadeThree": "#8e8e93",
        "ShadeTwo": "#b5b5b5",
        "activeTintOne": "#e26901",
        "background": "#fff",
        "border": "#C9C9CE",
        "card": "#fff",
        "inActiveTintOne": "#e26901",
        "primary": "#1b365d",
        "screenBackground": "#eeeeee",
        "text": "#A59F9A",
        "textFive": "#555555",
        "textFour": "#4C4C4C",
        "textThree": "#000",
        "textTwo": "#777777",
      }
    }
    dark={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#8e8e93",
        "fontSize": 26.250000000000004,
        "marginTop": 20.83275,
      }
    }
  />
</View>
`;

exports[`renders USetting when unchecked using Snapshots 1`] = `
<View
  colorScheme="light"
  colors={
    Object {
      "ShadeOne": "#eeeeee",
      "ShadeThree": "#8e8e93",
      "ShadeTwo": "#b5b5b5",
      "activeTintOne": "#e26901",
      "background": "#fff",
      "border": "#C9C9CE",
      "card": "#fff",
      "inActiveTintOne": "#e26901",
      "primary": "#1b365d",
      "screenBackground": "#eeeeee",
      "text": "#A59F9A",
      "textFive": "#555555",
      "textFour": "#4C4C4C",
      "textThree": "#000",
      "textTwo": "#777777",
    }
  }
  dark={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityLabel=" , , "
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 2,
        "borderTopLeftRadius": 2,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "overflow": "hidden",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      colorScheme="light"
      colors={
        Object {
          "ShadeOne": "#eeeeee",
          "ShadeThree": "#8e8e93",
          "ShadeTwo": "#b5b5b5",
          "activeTintOne": "#e26901",
          "background": "#fff",
          "border": "#C9C9CE",
          "card": "#fff",
          "inActiveTintOne": "#e26901",
          "primary": "#1b365d",
          "screenBackground": "#eeeeee",
          "text": "#A59F9A",
          "textFive": "#555555",
          "textFour": "#4C4C4C",
          "textThree": "#000",
          "textTwo": "#777777",
        }
      }
      dark={false}
      importantForAccessibility="yes"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#999",
            "fontSize": 30,
          },
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#1b365d",
            "borderRadius": 2,
            "borderWidth": 1,
            "color": "transparent",
            "fontSize": 37.6,
            "height": 47,
            "marginTop": 20.83275,
            "overflow": "hidden",
            "padding": 1.4000000000000001,
            "textAlign": "center",
            "textAlignVertical": "center",
            "width": 47,
          },
          Object {
            "fontFamily": "telldusicons",
          },
        ]
      }
    >
      checkmark
    </Text>
  </View>
  <Text
    allowFontScaling={false}
    colorScheme="light"
    colors={
      Object {
        "ShadeOne": "#eeeeee",
        "ShadeThree": "#8e8e93",
        "ShadeTwo": "#b5b5b5",
        "activeTintOne": "#e26901",
        "background": "#fff",
        "border": "#C9C9CE",
        "card": "#fff",
        "inActiveTintOne": "#e26901",
        "primary": "#1b365d",
        "screenBackground": "#eeeeee",
        "text": "#A59F9A",
        "textFive": "#555555",
        "textFour": "#4C4C4C",
        "textThree": "#000",
        "textTwo": "#777777",
      }
    }
    dark={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#8e8e93",
        "fontSize": 26.250000000000004,
        "marginTop": 20.83275,
      }
    }
  />
</View>
`;
